src/uct/ib/rc/base/rc_ep.c

```diff
@@ -310,15 +310,15 @@ ucs_arbiter_cb_result_t uct_rc_ep_process_pending(ucs_arbiter_t *arbiter,
         return UCS_ARBITER_CB_RESULT_NEXT_GROUP;
     } else {
         ep = ucs_container_of(ucs_arbiter_elem_group(elem), uct_rc_ep_t, arb_group);
-        if (! uct_rc_ep_has_tx_resources(ep)) {
-            /* No ep resources */
-            return UCS_ARBITER_CB_RESULT_DESCHED_GROUP;
-        } else {
-            /* No iface resources */
             iface = ucs_derived_of(ep->super.super.iface, uct_rc_iface_t);
-            ucs_assertv(!uct_rc_iface_has_tx_resources(iface),
-                        "pending callback returned error but send resources are available");
+        if (!uct_rc_iface_has_tx_resources(iface)) {
```

Comment:
need to fix indentation here?
and move line 312 inside 'else'

Reply:
> need to fix indentation here?

fixed

> and move line 312 inside 'else'

do you mean shift ```ep = ucs_container_of(ucs_arbiter_elem_group(elem), uct_rc_ep_t, arb_group)``` to line 319? ```ep``` is used to get ```iface``` - it will segfault if I shift it

Reply:
ok, right